node/src/main/scala/coop/rchain/node/instances/BlockProcessorInstance.scala

```diff
@@ -1,32 +1,24 @@
 package coop.rchain.node.instances
 
 import cats.effect.Concurrent
-import cats.effect.concurrent.{Deferred, Ref}
+import cats.effect.concurrent.Ref
 import cats.instances.list._
 import cats.syntax.all._
 import coop.rchain.casper.blocks.BlockProcessor
-import coop.rchain.casper.blocks.proposer.{BugError, ProposeResult}
 import coop.rchain.casper.protocol.BlockMessage
-import coop.rchain.casper.{
-  BlockError,
-  BlockStatus,
-  Casper,
-  PrettyPrinter,
-  ValidBlock,
-  ValidBlockProcessing
-}
+import coop.rchain.casper.{Casper, PrettyPrinter, ValidBlockProcessing}
 import coop.rchain.models.BlockHash.BlockHash
+import coop.rchain.node.NodeRuntime.ProposeFunction
 import coop.rchain.shared.Log
-import fs2.concurrent.Queue
 import fs2.Stream
+import fs2.concurrent.Queue
 
 object BlockProcessorInstance {
   def create[F[_]: Concurrent: Log](
       blocksQueue: Queue[F, (Casper[F], BlockMessage)],
       blockProcessor: BlockProcessor[F],
       state: Ref[F, Set[BlockHash]],
-      proposerQueue: Queue[F, (Casper[F], Deferred[F, Option[Int]])],
-      autopropose: Boolean
+      triggerProposeF: Option[ProposeFunction[F]]
```

Comment:
This should close https://github.com/rchain/rchain/issues/3308

Reply:
Is the result of the `ProposeFunction` used anywhere?

Reply:
Thanks! I did not finish work actually, ptal https://github.com/rchain/rchain/pull/3312/commits/862bc47689cb8753af67088821ac8ebee9a663a0

Yes, `ProposeFunction` should be defined in casper module and return value is used propose API https://github.com/rchain/rchain/pull/3312/commits/862bc47689cb8753af67088821ac8ebee9a663a0#diff-71a7b671734067220382fb1243b725e73f70e08ad9e78d93c6bdcd23f3d9fb84R30